wallet/src/repository/wallet/mod.rs

```
@@ -648,116 +570,96 @@ where
-        let txn_id = state.transaction_next_id;
-        let old_balance = state.balance;
-        let mut batch = self.db.batch();
-
-        if let Some(model::KeyBalance { amount, .. }) = state.utxo_set.get(&out_ptr).cloned() {
-            let new_balance = old_balance
-                .checked_rem(amount)
-                .ok_or_else(|| Error::BalanceUnderflow)?;
-            let mut db_utxo_set: model::UtxoSet = self.db.get(&keys::account_utxo_set(account))?;
-            let txn_next_id = txn_id
-                .checked_add(1)
-                .ok_or_else(|| Error::TransactionValueOverflow)?;
-
-            db_utxo_set.remove(&out_ptr);
-
-            batch.put(&keys::transaction_value(account, txn_id), amount)?;
-            batch.put(
-                keys::transaction_entry(account, txn_id),
-                model::TransactionEntry::Debit,
-            )?;
-            batch.put(
-                keys::transaction_type(account, txn_id),
-                model::TransactionType::ValueTransfer,
-            )?;
-            batch.put(
-                keys::transaction_timestamp(account, txn_id),
-                convert_block_epoch_to_timestamp(block.epoch),
-            )?;
-            batch.put(keys::transaction_block(account, txn_id), block)?;
-            batch.put(keys::account_balance(account), new_balance)?;
-            batch.put(keys::account_utxo_set(account), db_utxo_set)?;
-            batch.put(keys::transaction_next_id(account), txn_next_id)?;
-            batch.put(keys::transactions_index(txn_hash), txn_id)?;
-            batch.put(keys::transaction_hash(account, txn_id), txn_hash)?;
-
-            self.db.write(batch)?;
-
-            state.transaction_next_id = txn_next_id;
-            state.balance = new_balance;
-            state.utxo_set.remove(&out_ptr);
+        let mut db_utxo_set: model::UtxoSet = self
+            .db
+            .get(&keys::account_utxo_set(account))
+            .unwrap_or_default();
```

`unwrap_or_default` ignores any errors and creates a new variable. Maybe it would be better to use `get_opt(...)?.unwrap_or_default()`, so errors are handled with `?`.

---

It was something from the inherited code. The `unwrap_or_default` is used when the UXTO set was never initialized and an empty one is created.